core/src/main/java/org/bitcoinj/evolution/AbstractQuorumState.java

```diff
@@ -530,7 +526,7 @@ public boolean isDeterministicMNsSporkActive() {
     }
 
     public void addEventListeners(AbstractBlockChain blockChain, PeerGroup peerGroup) {
-        blockChain.addNewBestBlockListener(Threading.SAME_THREAD, newBestBlockListener);
+        blockChain.addNewBestBlockListener(newBestBlockListener);
```

Comment:
If `newBestBlockListener` is running in SAME_THREAD, then it will be running under an `AbstractBlockChain` lock which may be a cause of the deadlock I am trying to fix.  

This is the only listener running in the same thread.